app/src/main/java/com/danosoftware/galaxyforce/services/googleplay/GooglePlayServices.java

```diff
@@ -0,0 +1,225 @@
+        // this will in-turn pass the result back to our handleSignInResult()
+        mActivity.startActivityForResult(intent, RC_SIGN_IN);
+    }
+
+    public void handleSignInResult(Task<GoogleSignInAccount> completedTask) {
+
+//        GoogleSignInResult result = Auth.GoogleSignInApi.getSignInResultFromIntent(data);
+//
+//        result.
+//        if (result.isSuccess()) {
+//            // The signed in account is stored in the result.
+//            GoogleSignInAccount signedInAccount = result.getSignInAccount();
+////                GamesClient gamesClient = Games.getGamesClient(this, signedInAccount);
+////                gamesClient.setViewForPopups(this.findViewById(android.R.id.content));
+//        } else {
+//            String message = result.getStatus().getStatusMessage();
+//            if (message == null || message.isEmpty()) {
+//                message = "Unknown";
+//            }
+////                new AlertDialog.Builder(this).setMessage(message)
+////                        .setNeutralButton(android.R.string.ok, null).show();
+//        }
+
+
+
+
+//        if(completedTask.isSuccessful()) {
+//            GoogleSignInAccount signedInAccount = completedTask.getResult();
+//            Log.i(ACTIVITY_TAG, "signInResult:success");
+//            onConnected(signedInAccount);
+//        } else {
+//            String message = completedTask.getStatus().getStatusMessage();
+//            if (message == null || message.isEmpty()) {
+//                message = "Unknown";
+//            }
+//                new AlertDialog.Builder(mActivity).setMessage(message)
+//                        .setNeutralButton(android.R.string.ok, null).show();
+//        }
+//            Exception exception = completedTask.getException();
+//            if (exception instanceof ApiException) {
+//                final ApiException apiException = (ApiException) exception;
+//                final int exceptionCode = apiException.getStatusCode();
+//                final String exceptionCause = CommonStatusCodes.getStatusCodeString(exceptionCode);
+//                Log.w(ACTIVITY_TAG, String.format(
+//                        "signInResult:failure code={}. reason={}",
+//                        exceptionCode,
+//                        exceptionCause));
+//                connectedState = ConnectedState.DISCONNECTED;
+//            } else {
+//                Log.w(ACTIVITY_TAG, String.format(
+//                        "signInResult:failure reason={}",
+//                        exception.getMessage()));
+//            }
+//        }
+
+
+        try {
+            GoogleSignInAccount account = completedTask.getResult(ApiException.class);
+            Log.i(ACTIVITY_TAG, "signInResult:success");
+            onConnected(account);
+        } catch (ApiException e) {
+            Log.w(ACTIVITY_TAG, "signInResult:failed code=" + e.getStatusCode());
+            connectedState = ConnectedState.DISCONNECTED;
+//            signInSilently();
```

Comment:
Recommended process is:
```
if (requestCode == RC_SIGN_IN) {
    GoogleSignInResult result = Auth.GoogleSignInApi.getSignInResultFromIntent(data);
    if (result.isSuccess()) {
      // The signed in account is stored in the result.
      GoogleSignInAccount signedInAccount = result.getSignInAccount();
    } else {
      String message = result.getStatus().getStatusMessage();
      if (message == null || message.isEmpty()) {
        message = getString(R.string.signin_other_error);
      }
      new AlertDialog.Builder(this).setMessage(message)
          .setNeutralButton(android.R.string.ok, null).show();
    }
  }
```

See: https://developers.google.com/games/services/android/signin